server/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -326,11 +326,7 @@ public static FileStore getFileStore(final Path path) throws IOException {
 
     public static long getUsableSpace(Path path) throws IOException {
         long freeSpaceInBytes = Environment.getFileStore(path).getUsableSpace();
-
-        /* See: https://bugs.openjdk.java.net/browse/JDK-8162520 */
-        if (freeSpaceInBytes < 0) {
-            freeSpaceInBytes = Long.MAX_VALUE;
-        }
```

Comment:
The [linked issue](https://bugs.openjdk.org/browse/JDK-8162520) is resolved in Java 14. Since 8.0 requires Java 17, I think this is no longer needed.